omnibus/package-scripts/iot-agent/prerm

```diff
@@ -36,6 +36,11 @@ deregister_agent()
     fi
 }
 
+if [ "$DISTRIBUTION" = "Darwin" ]; then
+	echo "The IoT Agent is not supported on MacOS"
+	exit 0
+fi
+
```

Comment:
I think there's no need to add anything for MacOS in the `prerm`, it's not shipped in the `.pkg` (only the `preinst` and `postinst` are).
See: https://github.com/DataDog/omnibus-ruby/blob/94d7bdf36ca2b4adb8167a3d54bb30732fefce47/lib/omnibus/packagers/pkg.rb#L20-L27